internal/tagencoding/tagencoding.go

```diff
@@ -38,14 +32,11 @@ func (vb *Values) growIfRequired(expected int) {
 }
 
 func (vb *Values) WriteValue(v []byte) {
-	length := len(v)
-	vb.growIfRequired(sizeOfUint16 + length)
+	length := len(v) & 0xff
+	vb.growIfRequired(1 + length)
 
 	// writing length of v
-	bytes := *(*[2]byte)(unsafe.Pointer(&length))
-	vb.Buffer[vb.WriteIndex] = bytes[0]
-	vb.WriteIndex++
-	vb.Buffer[vb.WriteIndex] = bytes[1]
+	vb.Buffer[vb.WriteIndex] = byte(length)
```

Comment:
This doesn't seem right though as we are writing into the buffer a max of 0xffff, unless am mistaken.

Reply:
To keep parity with the old code, we need two different vb.WriteIndex additions or just one but not truncating length to 0xff only, please advise on this.